Tests/MongoSwiftTests/ConnectionStringTests.swift

```diff
@@ -262,4 +262,89 @@ final class ConnectionStringTests: MongoSwiftTestCase {
             expect(try client.listDatabases().wait()).to(throwError())
         }
     }
+
+    func testHeartbeatFrequencyMSOption() throws {
+        // option is set correctly from options struct
+        let opts = MongoClientOptions(heartbeatFrequencyMS: 50000)
+        let connStr1 = try ConnectionString("mongodb://localhost:27017", options: opts)
+        expect(connStr1.options?["heartbeatfrequencyms"]?.int32Value).to(equal(50000))
+
+        // option is parsed correctly from string
+        let connStr2 = try ConnectionString("mongodb://localhost:27017/?heartbeatFrequencyMS=50000")
+        expect(connStr2.options?["heartbeatfrequencyms"]?.int32Value).to(equal(50000))
+
+        // options struct overrides string
+        let connStr3 = try ConnectionString("mongodb://localhost:27017/?heartbeatFrequencyMS=20000", options: opts)
+        expect(connStr3.options?["heartbeatfrequencyms"]?.int32Value).to(equal(50000))
+
+        let tooSmall = 10
+        expect(try ConnectionString("mongodb://localhost:27017/?heartbeatFrequencyMS=\(tooSmall)"))
+            .to(throwError(errorType: MongoError.InvalidArgumentError.self))
+
+        expect(try ConnectionString(
+            "mongodb://localhost:27017",
+            options: MongoClientOptions(heartbeatFrequencyMS: tooSmall)
+        )).to(throwError(errorType: MongoError.InvalidArgumentError.self))
+
+        guard !MongoSwiftTestCase.is32Bit else {
+            print("Skipping remainder of test, only supported on 64-bit platforms")
+            return
+        }
+
+        let tooLarge = Int(Int32.max) + 1
+        expect(try ConnectionString(
+            "mongodb://localhost:27017",
+            options: MongoClientOptions(heartbeatFrequencyMS: tooLarge)
+        )).to(throwError(errorType: MongoError.InvalidArgumentError.self))
+
+        expect(try ConnectionString("mongodb://localhost:27017/?heartbeatFrequencyMS=\(tooLarge)"))
+            .to(throwError(errorType: MongoError.InvalidArgumentError.self))
+    }
+
+    fileprivate class HeartbeatWatcher: SDAMEventHandler {
+        fileprivate var started: [Date] = []
+        fileprivate var succeeded: [Date] = []
+
+        // listen for TopologyDescriptionChanged events and continually record the latest description we've seen.
+        fileprivate func handleSDAMEvent(_ event: SDAMEvent) {
+            switch event {
+            case .serverHeartbeatStarted:
+                self.started.append(Date())
+            case .serverHeartbeatSucceeded:
+                self.succeeded.append(Date())
+            default:
+                return
+            }
+        }
+
+        fileprivate init() {}
+    }
+
+    func testHeartbeatFrequencyMSWithMonitoring() throws {
+        guard MongoSwiftTestCase.topologyType == .single else {
```

Comment:
rather than worrying about heartbeats to multiple servers it was simpler to just run this against standalones.